### PR TITLE
Support srv-host records

### LINKED ISSUE #1823: Support for srv-host records in dnsmasq
## The problem

Need to configure dnsmasq to support a local domain controller as explained [here](https://mroach.com/2018/03/active-directory-dns-and-a-ubiquiti-edgerouter/)
However, the addon has no ability to add srv-host records.

## Environment

- Add-on with the issue: DNSmasq
- Add-on release with the issue:  1.3
- Last working add-on release (if known): Never
- Operating environment (OS/Supervised): hassos

## Problem-relevant configuration

N/A

## Traceback/Error logs

N/A

### PR DESCRIPTION
**Pull request by ocalvo**
Fixes #1823

### CODE REVIEW

File: dnsmasq/DOCS.md [hunk 1]
```diff
@@ -32,6 +32,15 @@ forwards:
 hosts:
   - host: home.mydomain.io
     ip: 192.168.1.10
+srv-hosts:
```
Comment (pvizeli) [inline dnsmasq/DOCS.md:35]: Just call this `srv`

File: dnsmasq/DOCS.md [hunk 2]
```diff
@@ -32,6 +32,15 @@ forwards:
...
+  - srv: _ldap._tcp.pdc._msdcs.mydomain.io
+    host: dc.mydomain.io
+    port: 389
+    priority: 0
+    weight: 100
+extra-options:
+  - option: "domain=mydomain.io"
+  - option: "mx-host=mx.mydomain.io"
```
Comment (pvizeli) [inline dnsmasq/DOCS.md:43]: That is a bit dump. Please make a better logic and expose not raw config options out

File: dnsmasq/data/run.sh [hunk 3]
```diff
@@ -26,6 +26,24 @@ for host in $(bashio::config 'hosts|keys'); do
...
 
+# Create static srv-hosts
+for srvhost in $(bashio::config 'srv-hosts|keys'); do
+    SRV=$(bashio::config "srv-hosts[${srvhost}].srv")
+    HOST=$(bashio::config "srv-hosts[${srvhost}].host")
+    PORT=$(bashio::config "srv-hosts[${srvhost}].port")
+    PRIORITY=$(bashio::config "srv-hosts[${srvhost}].priority")
+    WEIGHT=$(bashio::config "srv-hosts[${srvhost}].weight")
+
+    echo "srv-host=${SRV},${HOST},${PORT},${PRIORITY},${WEIGHT}" >> "${CONFIG}"
```
Comment (pvizeli) [inline dnsmasq/data/run.sh:37]: We do not use anymore echo or sed to generate config files. Please make a template at `/us/share/tempio` and generate the config from that template on runtime with bashio. I.e. zwave-js do the same or homematics use that also often.

**Review by pvizeli — CHANGES_REQUESTED** [review]

File: dnsmasq/DOCS.md (hunk 2 above)
Comment (ocalvo) [reply dnsmasq/DOCS.md:43]: Why we don't want expose all raw options for dnsmasq?

File: dnsmasq/data/run.sh (hunk 3 above)
Comment (ocalvo) [reply dnsmasq/data/run.sh:37]: Can you point me to the docs on how to use bashio templates, specifically how to use a for-loop?

File: dnsmasq/DOCS.md (hunk 1 above)
Comment (ocalvo) [reply dnsmasq/DOCS.md:35]: Rename it to "services"

File: dnsmasq/data/run.sh (hunk 3 above)
Comment (pvizeli) [reply dnsmasq/data/run.sh:37]: https://github.com/home-assistant/tempio

File: dnsmasq/rootfs/usr/share/tempio/dnsmasq.config [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Automatically generated do not edit
+
+# Default forward servers
+{{range .defaults}}
```
Comment (pvizeli) [inline dnsmasq/rootfs/usr/share/tempio/dnsmasq.config:4]: Let's use the same style `{{ ... }}` with an space between the `{{` or `}}`.

File: dnsmasq/config.json [hunk 5]
```diff
@@ -30,6 +32,20 @@
...
+    "services": [
+      {
+        "srv": "str",
+        "host": "str",
+        "port": "str",
+        "priority": "int",
+        "weight": "int"
+      }
+    ],
+    "extra_options": [
```
Comment (pvizeli) [inline dnsmasq/config.json:45]: I don't like common options that get not validate and end up in a not working state. Can you not expose on/off function for that options which you need?

**Comment by pvizeli** [comment]
The Apparmor Profile need things like:

```
  /etc/s6/** ix,
  /run/s6/** ix,
  /etc/services.d/** rwix,
  /etc/cont-init.d/** rwix,
  /etc/cont-finish.d/** rwix,

  /run/** rwk,
```

The existing run part can be replaced with the run above.

**Comment by ocalvo** [comment]
> The Apparmor Profile need things like:
> 
> ```
>   /etc/s6/** ix,
>   /run/s6/** ix,
>   /etc/services.d/** rwix,
>   /etc/cont-init.d/** rwix,
>   /etc/cont-finish.d/** rwix,
> 
>   /run/** rwk,
> ```
> 
> The existing run part can be replaced with the run above.
> 
> 

I don't understand, what is apparmor? Is this an example addon?

**Comment by pvizeli** [comment]
This add-on use apparmor, update the profile with my suggestion and it should run

**Review by pvizeli — APPROVED** [review]
Update changelog and bump the version and we can merge it

**Comment by ocalvo** [comment]
> Update changelog and bump the version and we can merge it

Version bumper to 1.4 and updated change log.

**Pull request merged**